alloy/src/main/java/com/liferay/faces/alloy/taghandler/LoadConstants.java

```diff
@@ -90,37 +91,70 @@ public LoadConstants(TagConfig config) throws Exception {
 	}
 
 	@Override
-	public void apply(FaceletContext facesContext, UIComponent parentUIComponent) throws IOException {
+	public void apply(FaceletContext facletContext, UIComponent parentUIComponent) throws IOException {
+
+		Cache<String, Map<String, Object>> constantCache = LoadConstants.constantCache;
+
+		// First check without locking (not yet thread-safe)
+		if (constantCache == null) {
+
+			// Second check with locking (thread-safe)
```

Comment:
@stiemannkj1 Please move this comment "Second check with locking (thread-safe)" to just above `if (constantCache == null) {` in order to stay consistent with usage of the double-check idom elsewhere in the codebase. For example: https://github.com/liferay/liferay-faces-bridge-api/blob/master/bridge-api/src/main/java/javax/portlet/faces/GenericFacesPortlet.java#L795